apps/mobile-wallet/src/persistent-storage/wallet.ts

```diff
@@ -311,3 +260,108 @@ export const dangerouslyExportWalletMnemonic = async (): Promise<string> => {
 
   return dangerouslyConvertUint8ArrayMnemonicToString(parsedDecryptedMnemonic)
 }
+
+export const getAddressAsymetricKey = async (addressHash: AddressHash, keyType: 'public' | 'private') => {
+  const storageKey = (keyType === 'public' ? ADDRESS_PUB_KEY_PREFIX : ADDRESS_PRIV_KEY_PREFIX) + addressHash
+  let key = await SecureStore.getItemAsync(storageKey, defaultSecureStoreConfig)
+
+  if (!key) {
+    const { addresses } = await getStoredWallet(`Could not get address ${keyType} key, wallet metadata not found`)
+    const address = addresses.find((address) => address.hash === addressHash)
+
+    if (!address) throw new Error(`Could not get address ${keyType} key, address metadata not found`)
+
+    await generateAndStoreAddressKeypairForIndex(address.index)
+    key = await SecureStore.getItemAsync(storageKey, defaultSecureStoreConfig)
+
+    if (!key) throw new Error(`Could not generate address ${keyType} key`)
+  }
+
+  return key
+}
+
+const generateWalletMetadata = (name: string, firstAddressHash: string, isMnemonicBackedUp = false) => ({
```

Comment:
Very nit: I'd call this `generateInitialWalletMetadata`

Reply:
I think the `generate` prefix is enough :)

Reply:
In general, you can generate something with isn't the initial item. But ok, very nit!